compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -15784,8 +15799,9 @@ OMR::Z::TreeEvaluator::arraycmpSIMDHelper(TR::Node *node,
       if(isArrayCmpLen)
          {
          // Return 0-based index of first non-matching element
-         // resultReg - lastByteIndexReg = number of elements compared before the last loop
+         // (resultReg - 1) - lastByteIndexReg = number of elements compared before the last loop
```

Comment:
Can you explain why we subtract by 1 here (Which we were not doing before

Reply:
sure, previously, it was subtracting one from lastByteIndexReg then moving its value to the result register in removed line 15730. I reordered code copy lastByteIndexReg to resultRegister before the subtraction therefore need to subtract 1 from it here. It was because I needed the result register to have full length for short circuit operation.

Reply:
Thanks for clarification. It makes sense. I can not think better sequence top of my head for such, but would try to see if there are some better sequence we can generate here (Reason here is that, short circuit is not the common case), We did introduce the extra operation for the commonly executed cases with arraycmpLen.